tests/e2e/kcs/test_selinux_relabel_solution.py

```diff
@@ -330,3 +330,114 @@ def test_selinux_relabel_for_existing_pvc(
+
+        """
+        # Create storageclass with security context
+        self.ocp_project = ocp.OCP(
+            kind=constants.NAMESPACE, namespace=constants.OPENSHIFT_STORAGE_NAMESPACE
+        )
+        sc_name = "ocs-storagecluster-cephfs-selinux-relabel"
+        self.storage_class = storageclass_factory(
+            sc_name=sc_name,
+            interface=constants.CEPHFILESYSTEM,
+            kernelMountOptions='context="system_u:object_r:container_file_t:s0"',
+        )
+
+        # Create PVC using new storageclass
+        self.pvc_obj = pvc_factory(
+            project=self.ocp_project,
+            storageclass=self.storage_class,
+            size="20",
+        )
+        log.info(f"PVC {self.pvc_obj.name} created")
+
+        # Create service_account to get privilege for deployment pods
+        self.service_account_obj = service_account_factory(
+            project=self.ocp_project,
+        )
+
+        # Create deployment pod
+        self.pod_obj = self.create_deploymentconfig_pod(
+            command=["/opt/multiple_files.sh"],
+            command_args=[f"{copies}", "/mnt"],
+        )
+
+        log.info(f"pod {self.pod_obj.name} created")
+        self.pod_selector = self.pod_obj.labels.get(constants.DEPLOYMENTCONFIG)
+
+        # Get the md5sum of some random files
+        data_path = f"{constants.FLEXY_MNT_CONTAINER_DIR}"
+        num_of_files = random.randint(3, 9)
+        pod = ocp.OCP(kind="pod", namespace=constants.OPENSHIFT_STORAGE_NAMESPACE)
+        random_files = pod.exec_oc_cmd(
+            f"exec -it {self.pod_obj.name} -- /bin/bash"
+            f' -c "find {data_path} -type f | "shuf" -n {num_of_files}"',
+            timeout=300,
+        )
+        random_files = random_files.split()
+        log.info(f"files are {random_files}")
+        initial_md5sum = []
+        for file_path in random_files:
+            md5sum = res_pod.cal_md5sum(
+                pod_obj=self.pod_obj,
+                file_name=file_path,
+            )
+            initial_md5sum.append(md5sum)
+
+        # Check data integrity and Get time for pod restart
+        self.pod_obj.delete(wait=True)
+        self.pod_obj = self.get_app_pod_obj()
+        try:
+            wait_for_pods_to_be_running(
+                pod_names=[self.pod_obj.name], timeout=600, sleep=15
+            )
+        except CommandFailed:
+            log.exception(f"Pod {self.pod_obj.name} didn't reach to running state")
+
+        pod_restart_time_after_fix = self.get_pod_start_time(pod_name=self.pod_obj.name)
+        log.info(f"Time taken by pod to restart is {pod_restart_time_after_fix}")
+        final_md5sum = []
+        for file_path in random_files:
+            md5sum = res_pod.cal_md5sum(
+                pod_obj=self.pod_obj,
+                file_name=file_path,
+            )
+            final_md5sum.append(md5sum)
+        assert initial_md5sum == final_md5sum, "Data integrity failed after on new PVC."
```

Comment:
correct the assert message and also print the pvc name for which the integrity check failed -  "Data integrity failed for pvc: {pvc.name}"